File: mp-checkout/marketplace/nodejs/payment_button.js
var MP = require ("mercadopago"),

exports.run = function (req, res) {
    var mp = new MP ("SELLER_AT"); // seller access_token

    var preference = {
        "items": [
            {
                "title": "Item title",
                "description": "Description",
                "quantity": 1,
                "unit_price": 50,
                "currency_id": "CURRENCY_ID",
                "picture_url": "https://www.mercadopago.com/org-img/MP3/home/logomp3.gif"
            }
        ],
        "marketplace_fee": 2.29 // fee to collect
    };

    mp.createPreference (preference, function (err, data){
        if (err) {
            res.send (err);
        } else {
            res.render ("button", {"preference": data});
        }
    });
};

/* view button.jade
!!! 5
html
    head
        Pay

    body
        (href=preference.response.init_point) Pay
*/

File: mp-checkout/marketplace/ruby/payment_button.rb
require 'rubygems'
require 'rack'
require 'mercadopago.rb'

class Button
	def call(env)
		mp = MercadoPago.new('SELLER_AT') # seller access_token
		preferenceData = {
			"items" => [
				[
					"title" => "Item title",
					"description" => "Description",
					"quantity" => 1,
					"unit_price" => 50,
					"currency_id" => "CURRENCY_ID",
					"picture_url" => "https://www.mercadopago.com/org-img/MP3/home/logomp3.gif"
				]
			],
			"marketplace_fee" => 2.29 # fee to collect
		}
		preference = mp.create_preference(preferenceData)

		html =  '<!doctype html>
				<html>
					<head>
						<title>Pay</title>
					</head>
					<body>
						<a href="' + preference['response']['init_point'] + '">Pay</a>
					</body>
				</html>'

		return [200, {'Content-Type' => 'text/html'}, [html]]
	end
end

Rack::Handler::WEBrick.run(Button.new, :Port => 9000)